src/mongo/integration/standalone/dbclient_test.cpp

```diff
@@ -1377,6 +1377,14 @@ TEST_F(DBClientTest, CreateHashedIndex) {
     c->createIndex(TEST_NS, IndexSpec().addKey("aField", IndexSpec::kIndexTypeHashed));
 }
 
+TEST_F(DBClientTest, CreatePartialIndex) {
+    if (serverGTE(c.get(), 3, 1)) {
+        c->createIndex(
+            TEST_NS,
+            IndexSpec().addKey("aField").partialFilterExpression(BSON("aField" << GT << 5)));
+    }
```

Comment:
can you check with getIndexInfos that the server's index spec has the expected partialFilterExpression field?